modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/UndoRedo.js

```diff
@@ -0,0 +1,214 @@
+				nextStep.titleInputComponent,
+				nextStep.selectedLanguageId
+			);
+			descriptionInputComponent.updateInputLanguage(
+				nextStep.descriptionInputComponent,
+				nextStep.selectedLanguageId
+			);
+			friendlyURLInputComponent.updateInputLanguage(
+				nextStep.friendlyURLInputComponent,
+				nextStep.selectedLanguageId
+			);
+			titleInputComponent.updateInput(nextStep.titleInputComponent);
+			descriptionInputComponent.updateInput(
+				nextStep.descriptionInputComponent
+			);
+			friendlyURLInputComponent.updateInput(
+				nextStep.friendlyURLInputComponent
+			);
+		}
+		setState({
+			defaultLanguageId: nextStep.defaultLanguageId,
+			history,
+			selectedLanguageId: nextStep.selectedLanguageId,
+			step: newStep,
+		});
+	};
+
+	const handleAutoSave = useCallback(
+		({fieldName}) => {
+			const defaultLanguageIdInput = document.getElementById(
+				`${namespace}defaultLanguageId`
+			);
+
+			const descriptionInputComponent = Liferay.componentReady(
+				`${namespace}${META_FIELD_NAMES.description}`
+			);
+
+			const titleInputComponent = Liferay.componentReady(
+				`${namespace}${META_FIELD_NAMES.title}`
+			);
+
+			const friendlyURLInputComponent = Liferay.componentReady(
+				`${namespace}${META_FIELD_NAMES.friendlyURL}`
+			);
+
+			const selectedLanguageIdInput = document.getElementById(
+				`${namespace}languageId`
+			);
+
+			Promise.all([
+				descriptionInputComponent,
+				titleInputComponent,
+				friendlyURLInputComponent,
+			]).then(
+				([
+					descriptionInputComponent,
+					titleInputComponent,
+					friendlyURLInputComponent,
+				]) => {
+					const newHistory = {
+						defaultLanguageId: defaultLanguageIdInput.value,
+						descriptionInputComponent: descriptionInputComponent.getValue(
+							selectedLanguageId
+						),
+						friendlyURLInputComponent: friendlyURLInputComponent.getValue(
+							selectedLanguageId
+						),
+						name: fieldName,
+						selectedLanguageId: selectedLanguageIdInput.value,
+						titleInputComponent: titleInputComponent.getValue(
+							selectedLanguageId
+						),
+					};
+
+					setState({
+						defaultLanguageId: defaultLanguageIdInput.value,
+						history: [...history.slice(0, step + 1), newHistory],
+						selectedLanguageId: selectedLanguageIdInput.value,
+						step: step + 1,
+					});
+				}
+			);
+		},
+		[history, namespace, selectedLanguageId, step]
+	);
+
+	const localeChangeHandler = useCallback(
+		(event) => {
+			const selectedLanguageId = event.item.getAttribute('data-value');
+
+			const selectedLanguageIdInput = document.getElementById(
+				`${namespace}languageId`
+			);
+
+			selectedLanguageIdInput.value = selectedLanguageId;
+
+			Liferay.fire('autoSave', {fieldName: 'Locale Change'});
+		},
+		[namespace]
+	);
+
+	useEffect(() => {
+		Liferay.after('inputLocalized:localeChanged', localeChangeHandler);
+
+		return () => {
+			Liferay.detach('inputLocalized:localeChanged', localeChangeHandler);
```

Comment:
What is there are multiple localized inputs on the page? Wouldn't they all trigger these events?

Reply:
The inputLocalized:LocaleChanged event is fired when the locale is changed from a language selector, It doesn't fire multiple times for multiple inputLocalized fields, however all of them listen to it as well.